File: LoginIS/LoginSystem.py
import mysql.connector

# 连接数据库
mydb = mysql.connector.connect(
    host="localhost",
    user="root",
    password="10514",
    database="longin"
)

mycursor = mydb.cursor()

def register():
    username = input("请输入用户名：")
    password = input("请输入密码：")

    # 检查用户名是否已存在
    sql = "SELECT * FROM users WHERE username = %s"
    val = (username,)
    mycursor.execute(sql, val)
    result = mycursor.fetchone()

    if result:
        print("用户名已存在，请重新输入。")
        return

    # 插入新用户
    sql = "INSERT INTO users (username, password) VALUES (%s, %s)"
    val = (username, password)
    mycursor.execute(sql, val)
    mydb.commit()
    print("注册成功！")

def login():
    username = input("请输入用户名：")
    password = input("请输入密码：")

    # 检查用户名和密码是否匹配
    sql = "SELECT * FROM users WHERE username = %s AND password = %s"
    val = (username, password)
    mycursor.execute(sql, val)
    result = mycursor.fetchone()

    if result:
        print("登录成功！")
    else:
        print("用户名或密码错误，请重新输入。")

# 主程序
while True:
    choice = input("1. 注册  2. 登录  3. 退出\n请选择：")
    if choice == '1':
        register()
    elif choice == '2':
        login()
    elif choice == '3':
        break
    else:
        print("无效的选择，请重新输入。")

File: LoginIS/LoginSystem.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class LoginSystem {

    // 数据库连接参数
    private static final String DB_URL = "jdbc:mysql://localhost:3306/login";
    private static final String USER = "root";
    private static final String PASS = "10514";

    /**
     * 验证用户登录信息的方法
     *
     * @param username 用户名
     * @param password 密码
     * @return 如果用户存在且密码匹配，返回 true；否则返回 false
     */
    public static boolean validateUser(String username, String password) {
        try (Connection conn = DriverManager.getConnection(DB_URL, USER, PASS);
             PreparedStatement stmt = conn.prepareStatement("SELECT * FROM users WHERE username =? AND password =?")) {

            stmt.setString(1, username);
            stmt.setString(2, password);

            try (ResultSet rs = stmt.executeQuery()) {
                return rs.next(); // 如果有结果，即用户存在且密码匹配，返回 true
            }
        } catch (SQLException e) {
            e.printStackTrace();
            return false; // 出现异常返回 false
        }
    }

    public static void main(String[] args) {
        String username = "your_entered_username";
        String password = "your_entered_password";

        if (validateUser(username, password)) {
            System.out.println("Login successful");
        } else {
            System.out.println("Login failed");
        }
    }
}